### PR TITLE
make sure all torch._numpy tests run on CI

### PR DESCRIPTION
**Pull request by ev-br**
- Add `if __name__ == "__main__": run_tests()` stanzas to test files in `torch_np` folder so that these tests run on CI
- Skip / xfail things smoked out by this change 
- remove a stray python file which should not have been added to tests in the first place.
- fix einsum if opt_einsum is present
- add skips for older numpies

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/torch_np/test_basic.py [hunk 1]
```diff
@@ -571,6 +572,7 @@ def test_ndarrays_to_tensors():
     assert isinstance(out[0][0], torch.Tensor)
 
 
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="requires cuda")
```
Comment (lezcano) [inline test/torch_np/test_basic.py:575]: There's a `skipIfNoCuda` (not real name, but you know) wrapper we use for this.

Comment (ev-br) [reply test/torch_np/test_basic.py:575]: https://github.com/pytorch/pytorch/blob/main/test/dynamo/test_misc.py#L2391 uses `skipif(not TEST_CUDA)`, and `TEST_CUDA` is precisely the same: https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_utils.py#L1062. Will change.

**Comment by ev-br** [comment]
torch_np tests all run on CI, remaining CI failures do not seem related.

**Review by lezcano — APPROVED** [review]
Great! Thank you!

File: test/torch_np/numpy_tests/lib/test_shape_base_.py [hunk 2]
```diff
@@ -94,6 +94,7 @@ def test_empty(self):
         actual = take_along_axis(a, ai, axis=1)
         assert_equal(actual.shape, ai.shape)
 
+    @pytest.mark.xfail(reason="https://github.com/pytorch/pytorch/pull/107875")
```
Comment (lezcano) [inline test/torch_np/numpy_tests/lib/test_shape_base_.py:97]: @guilhermeleobas if this PR lands before https://github.com/pytorch/pytorch/pull/108880#pullrequestreview-1618672157 you'll need to remove this xfail

**Comment by ev-br** [comment]
The offending test was doing essentially this:

```
(Pdb) p x
tensor(-2147483648, dtype=torch.int32)
(Pdb) p torch.iinfo(torch.int32).min
-2147483648
(Pdb) p torch.negative(x)
tensor(-2147483648, dtype=torch.int32)      # NB: equals `x`
(Pdb) p torch.negative(x.to(torch.int64))
tensor(2147483648)
```

under ASAN, this produced instead (from the CI logs):

```
    2023-09-11T23:10:59.0824026Z torch_np/numpy_tests/core/test_scalarmath.py::test_scalar_signed_integer_overflow[neg-i] /var/lib/jenkins/workspace/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp:290:3: runtime error: negation of -2147483648 cannot be represented in type 'scalar_t' (aka 'int'); cast to an unsigned type to negate this value to itself
    2023-09-11T23:10:59.0825372Z     #0 0x7f86347119f4 in void at::native::AVX2::execute_op<at::native::AVX2::neg_kernel(at::TensorIteratorBase&)::$_25::operator()() const::'lambda1'()::operator()() const::'lambda'(int)&, (void*)0>(char* restrict*, long const*, long, long, at::native::AVX2::neg_kernel(at::TensorIteratorBase&)::$_25::operator()() const::'lambda1'()::operator()() const::'lambda'(int)&) (/opt/conda/envs/py_3.9/lib/python3.9/site-packages/torch/lib/libtorch_cpu.so+0x140f49f4)
    2023-09-11T23:10:59.0826756Z     #1 0x7f863471164d in void at::native::AVX2::basic_loop<at::native::AVX2::neg_kernel(at::TensorIteratorBase&)::$_25::operator()() const::'lambda1'()::operator()() const::'lambda'(int)&>(char* restrict*, long const*, long, long, at::native::AVX2::neg_kernel(at::TensorIteratorBase&)::$_25::operator()() const::'lambda1'()::operator()() const::'lambda'(int)&) (/opt/conda/envs/py_3.9/lib/python3.9/site-packages/torch/lib/libtorch_cpu.so+0x140f464d)
    2023-09-11T23:10:59.0828284Z     #2 0x7f8634711e3e in at::native::AVX2::VectorizedLoop2d<at::native::AVX2::neg_kernel(at::TensorIteratorBase&)::$_25::operator()() const::'lambda1'()::operator()() const::'lambda
```

While the operation is actually UB in C, the python-level test:
- in numpy it checked that a RuntimeWarning is emitted on overflow
- in torch._numpy we xfailed it, since pytorch does not emit these warnings.

An xfailed test still runs, hence the CI failure. Therefore, the last commit changes `pytest.mark.xfail` to `pytest.mark.skip` --- let's see if it's enough to prevent it from running at all.

**Pull request closed without merging**